searvey/_ndbc_api.py

```diff
@@ -0,0 +1,182 @@
+from __future__ import annotations
+
+import functools
+import logging
+from typing import List
+from typing import Union
+
+import geopandas as gpd
+import multifutures
+import numpy as np
+import pandas as pd
+from ndbc_api import NdbcApi
+from shapely.geometry import MultiPolygon
+from shapely.geometry import Polygon
+
+from searvey._common import _resolve_end_date
+from searvey._common import _resolve_start_date
+from searvey.custom_types import DatetimeLike
+from searvey.utils import get_region
+
+logger = logging.getLogger(__name__)
+
+# Create an instance of NdbcApi
+ndbc_api = NdbcApi()
```

Comment:
@abdu558 sorry I missed this earlier, but is there any way to avoid having to create module level global object for the third party API? e.g. is it very expensive to create the `NdbcApi` on every call? Or can you use cached singleton (cache the return value of a function that creates the object), see `cache` in `functools`

@pmav99 do you have any pattern suggestion?

Reply:
But It wont have to be created on every call. It would be created once when its imported??

Reply:
I mean to say implicitly creating a thirdparty global object on importing a module is not a good idea, that's why I'd like it to be replace by some other pattern, such as cached function return or API creation on every call (if not expensive)